common/processes/platform/tiny_process_library/tiny_process_library.cpp

```diff
@@ -178,7 +178,33 @@ error::Error Process::AsyncWait(events::EventLoop &loop, AsyncWaitHandler handle
 	return error::NoError;
 }
 
+error::Error Process::AsyncWait(
+	events::EventLoop &loop, AsyncWaitHandler handler, chrono::nanoseconds timeout) {
+	timeout_timer_.reset(new events::Timer(loop));
+
+	auto err = AsyncWait(loop, handler);
+	if (err != error::NoError) {
+		return err;
+	}
+
```

Comment:
Why aren't we using here the our private `async_wait_data_` (and related lock, etc)? Do we support parallel calls to the two overloads of `AsyncWait` (with and without timer)?

Reply:
It's because only the members inside the `async` call in `SetupAsyncWait` are executed in a different thread, everything else is in the main thread. What's executed inside the `Post()` is also on the main thread, since it is executing on the event loop. So we don't need to worry about any members not explicitly mentioned inside the `async`.

Reply:
> Do we support parallel calls to the two overloads of `AsyncWait` (with and without timer)?

No, the locks are only there because we internally use a thread. Parallel calls are not allowed.